hummingbot/connector/gateway_EVM_AMM.py

```diff
@@ -426,9 +459,11 @@ async def _create_order(self,
             transaction_hash: str = order_result.get("txHash")
             nonce: int = order_result.get("nonce")
             gas_price: Decimal = Decimal(order_result.get("gasPrice"))
-            gas_limit: int = order_result.get("gasLimit")
-            gas_cost: int = order_result.get("gasCost")
-            tracked_order = self._in_flight_orders.get(order_id)
+            gas_limit: int = int(order_result.get("gasLimit"))
+            gas_cost: Decimal = Decimal(order_result.get("gasCost"))
+            gas_price_token: str = order_result.get("gasPriceToken")
+            tracked_order: GatewayInFlightOrder = self._in_flight_orders.get(order_id)
+            self.update_transaction_fees(gas_price_token, gas_cost)
```

Comment:
As per the above suggestion wrt using property.setter, we will have to initialise the new `TokenAmount` here.